dvc/repo/experiments/executor/base.py

```diff
@@ -464,6 +489,10 @@ def checkpoint_callback(
             exp_rev = cls.commit(
                 scm, exp_hash, exp_name=name, force=force, checkpoint=True
             )
+
+            git_remote = os.environ.get(DVC_EXP_AUTO_PUSH, None)
+            if git_remote:
+                cls._auto_push(git_remote, dvc, scm)
```

Comment:
This check and push still needs to be moved from `checkpoint_callback()` into the end of `commit()`. With the current behavior, the final commit at the end of `reproduce()` is not pushed.